pkg/manifest/manifest_test.go

```diff
@@ -195,38 +207,91 @@ func TestReadManifest_File(t *testing.T) {
 }
 
 func TestSetDefault(t *testing.T) {
-	t.Run("bundle docker tag set", func(t *testing.T) {
+	t.Run("bundle docker tag set on reference", func(t *testing.T) {
 		m := Manifest{
+			Name:      "mybun",
 			Version:   "1.2.3-beta.1",
-			BundleTag: "getporter/mybun:v1.2.3",
+			Reference: "getporter/mybun:v1.2.3",
 		}
 		err := m.SetDefaults()
 		require.NoError(t, err)
 		assert.Equal(t, "getporter/mybun:v1.2.3", m.BundleTag)
 		assert.Equal(t, "getporter/mybun-installer:v1.2.3", m.Image)
 	})
 
-	t.Run("bundle docker tag not set", func(t *testing.T) {
+	t.Run("bundle docker tag not set on reference", func(t *testing.T) {
 		m := Manifest{
+			Name:      "mybun",
 			Version:   "1.2.3-beta.1",
-			BundleTag: "getporter/mybun",
+			Reference: "getporter/mybun",
 		}
 		err := m.SetDefaults()
 		require.NoError(t, err)
 		assert.Equal(t, "getporter/mybun:v1.2.3-beta.1", m.BundleTag)
 		assert.Equal(t, "getporter/mybun-installer:v1.2.3-beta.1", m.Image)
 	})
 
-	t.Run("bundle tag includes registry with port", func(t *testing.T) {
+	t.Run("bundle reference includes registry with port", func(t *testing.T) {
 		m := Manifest{
+			Name:      "mybun",
 			Version:   "0.1.0",
-			BundleTag: "localhost:5000/missing-invocation-image",
+			Reference: "localhost:5000/missing-invocation-image",
 		}
 		err := m.SetDefaults()
 		require.NoError(t, err)
 		assert.Equal(t, "localhost:5000/missing-invocation-image:v0.1.0", m.BundleTag)
 		assert.Equal(t, "localhost:5000/missing-invocation-image-installer:v0.1.0", m.Image)
 	})
+
+	t.Run("registry provided, no reference", func(t *testing.T) {
+		m := Manifest{
+			Name:     "mybun",
+			Version:  "1.2.3-beta.1",
+			Registry: "getporter",
+		}
+		err := m.SetDefaults()
+		require.NoError(t, err)
+		assert.Equal(t, "getporter/mybun:v1.2.3-beta.1", m.BundleTag)
+		assert.Equal(t, "getporter/mybun-installer:v1.2.3-beta.1", m.Image)
+	})
+
+	t.Run("registry provided with org, no reference", func(t *testing.T) {
+		m := Manifest{
+			Name:     "mybun",
+			Version:  "1.2.3-beta.1",
+			Registry: "getporter/myorg",
+		}
+		err := m.SetDefaults()
+		require.NoError(t, err)
+		assert.Equal(t, "getporter/myorg/mybun:v1.2.3-beta.1", m.BundleTag)
+		assert.Equal(t, "getporter/myorg/mybun-installer:v1.2.3-beta.1", m.Image)
+	})
+
+	t.Run("registry and reference provided", func(t *testing.T) {
+		m := Manifest{
+			Name:      "mybun",
+			Version:   "1.2.3-beta.1",
+			Registry:  "myregistry/myorg",
+			Reference: "getporter/org/mybun:v1.2.3",
+		}
+		err := m.SetDefaults()
+		require.NoError(t, err)
+		assert.Equal(t, "getporter/org/mybun:v1.2.3", m.BundleTag)
```

Comment:
Again let's update Reference and have it hold the final value, and not modify BundleTag (deprecated).